### PR TITLE
Update UG with more comprehensive autocomplete explanations

### LINKED ISSUE #128: Update UG with more comprehensive autocomplete explanations

### PR DESCRIPTION
**Pull request by isaactodo**
Closes #128

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -213,16 +213,48 @@ Format: `exit`
...
+
+<div markdown="span" class="alert alert-info"> :notebook: **Note:** Autocomplete will **not** match subsequent words after the first word with command keywords! </div>
+
+#### Supported Fields
+Autocomplete currently supports the following fields with these prefixes:
+
+* Modules `m/` — Matches **all existing** module names.
+* Tags `t/` — Matches **all existing** tags.
+* Gender `g/` — Matches gender values: `male` or `female`.
+* File Paths `pa/` — Matches **all existing** archived file paths.
```
Comment (RadieonAjax) [inline docs/UserGuide.md:233]: Consolidating the fields to a table might make it more concise! Something like this:

| Prefix   | Field            | Description                                     |
|----------|------------------|-------------------------------------------------|
| `m/`     | Modules          | Matches **all existing** module names               |
| `t/`     | Tags             | Matches **all existing** tags                       |
| `g/`     | Gender           | Matches gender values: `male` or `female`       |
| `pa/`    | File Paths       | Matches **all existing** archived file paths        |

**Review by RadieonAjax — APPROVED** [review]
LGTM!

File: docs/UserGuide.md (hunk 1 above)
Comment (RadieonAjax) [reply docs/UserGuide.md:233]:
```
| Prefix   | Field            | Description                                     |
|----------|------------------|-------------------------------------------------|
| `m/`     | Modules          | Matches **all existing** module names               |
| `t/`     | Tags             | Matches **all existing** tags                       |
| `g/`     | Gender           | Matches gender values: `male` or `female`       |
| `pa/`    | File Paths       | Matches **all existing** archived file paths        |
```

In markdown form^

Comment (isaactodo) [reply docs/UserGuide.md]: Changed to a table!

**Pull request merged**